lib/adbc_connection.ex

```diff
@@ -407,15 +407,20 @@ defmodule Adbc.Connection do
     end
   end
 
-  defp merge_columns([]), do: []
+  defp merge_columns(chucked_results) do
+    Enum.zip_with(chucked_results, fn columns ->
+      Enum.reduce(columns, fn column, merged_column ->
+        merged_data =
+          case {is_list(merged_column.data), is_list(column.data)} do
+            {true, true} -> merged_column.data ++ column.data
+            {true, false} -> merged_column.data ++ [column.data]
+            {false, true} -> [merged_column.data] ++ column.data
+            {false, false} -> [merged_column.data] ++ [column.data]
+          end
```

Comment:
You could write this as:

```suggestion
          List.wrap(merged_column.data) ++ List.wrap(column.data)
```

But you can also always force it to be a list of refs, since we need to deal with lists of refs anyways.

Reply:
It is probably simpler to go with a list of refs, it means there is less conditional logic elsewhere. But as you prefer.

Reply:
Oh I didn't even know this function exists before!

Reply:
Nice, but yeah, if we always return a list of refs, we won't need it. :)

Reply:
It's done in e225123, and I'm going to release v0.6.0 :D